app/services/PaymentAPIService.scala

```diff
@@ -116,9 +83,8 @@ class PaymentAPIService(wsClient: WSClient, paymentAPIUrl: String) {
     queryStrings: Map[String, Seq[String]],
     email: Option[String],
     isTestUser: Boolean
-  )(implicit ec: ExecutionContext): Future[Boolean] = {
+  )(implicit ec: ExecutionContext): Future[Option[PaymentAPIResponse[PayPalError, PaypalExecuteSuccessResponse]]] = {
```

Comment:
If this was an `Either` instead of an `Option`, we could use `EitherT` here.

Reply:
as in EitherT[Future, PayPalError, PaypalExecuteSuccessResponse] ?